src/pngimage.cpp

```diff
@@ -158,9 +158,12 @@ namespace Exiv2 {
         long        count=0;
         const byte* p = bytes ;
         // header is \nsomething\n number\n hex
-        while ( count < 3 )
-            if ( *p++ == '\n' )
+        while ((count < 3) && (p - bytes < length)) {
```

Comment:
oh god. This code is scary. Were you able to make this code fail before trying to fix it? 

I could not understand what you are trying to do here (neither what the original code is doing).

Reply:
I have added some comments explaining that part of the code.